packages/fetch-core/src/fetchToken.js

```diff
@@ -12,7 +12,7 @@ export const fetchWithToken = (fetch, getAccessTokenInjected) => async (
   // eslint-disable-next-line
   let headers = new Headers();
   // eslint-disable-next-line
-  if (options.headers && !(options.headers instanceof Headers)) {
```

Comment:
This modif will create bugs

I think you want to do this 

 if (options.headers){
  headers =  !(options.headers instanceof Headers) ? new Headers(options.headers) : options.headers;
}